scripts/test-tabs.ts

```
@@ -0,0 +1,171 @@
+  const line = result.lines[0];
+  const fullText = line.spans.map(s => s.text).join('');
+  const expectedTabPos = Math.ceil((input.indexOf('\t') + 1) / 8) * 8;
+  console.log(`  Input: "${input.replace('\t', '\\t')}" -> Output: "${visualizeWhitespace(fullText)}" (len=${fullText.length})`);
+  console.log(`    Tab should expand to column ${expectedTabPos}, actual len after first word: ${fullText.indexOf(input.split('\t')[1])}`);
+}
+console.log();
+
+// Test 6: ptyToText output (plain text extraction)
+console.log(`${YELLOW}Test 6: ptyToText (strips ANSI, keeps spacing)${RESET}`);
+const test6Input = '\x1b[32mgreen\x1b[0m\ttext\there';
+console.log(`Input with ANSI: "${test6Input.replace(/\x1b/g, '\\e')}"`);
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)